src/components/DeleteCollective.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import withIntl from '../lib/withIntl';
+import { withUser } from './UserProvider';
+import { addDeleteCollectiveMutation, addDeleteUserCollectiveMutation } from '../graphql/mutations';
+import { H2, P } from './Text';
+import Container from './Container';
+import StyledButton from './StyledButton';
+import Modal from './Modal';
+import { Router } from '../server/pages';
+
+const getCollectiveType = type => {
+  switch (type) {
+    case 'ORGANIZATION':
+      return 'Organization';
+    case 'COLLECTIVE':
+      return 'Collective';
+    default:
+      return 'Account';
+  }
+};
+
+const DeleteCollective = ({ collective, deleteCollective, deleteUserCollective, logout }) => {
+  const collectiveType = getCollectiveType(collective.type);
+  const [showModal, setShowModal] = useState(false);
+  const [deleteStatus, setDeleteStatus] = useState({
+    deleting: false,
+    error: null,
+  });
+
+  const handleDeleteCollective = async () => {
+    try {
+      setDeleteStatus({ ...deleteStatus, deleting: true });
+      await deleteCollective(collective.id);
+      await Router.pushRoute(`/deleteCollective/confirmed?type=${collective.type}`);
+    } catch (err) {
+      console.error('>>> deleteUserCollective error: ', JSON.stringify(err));
+      const errorMsg = err.graphQLErrors && err.graphQLErrors[0] ? err.graphQLErrors[0].message : err.message;
+      setDeleteStatus({ deleting: false, error: errorMsg });
+    }
+  };
+
+  const handleDeleteUserCollective = async () => {
```

Comment:
`handleDeleteCollective`and `handleDeleteUserCollective` are not different enough to justify a different function.